### ISSUE TITLE
async/await React Native

### ISSUE THREAD

**Issue by hadnet** [BrasilAPI/cep-promise#223]
Quando tento usar async/await diretamente, lança o erro:

```ts
 Possible Unhandled Promise Rejection (id: 0):
TypeError: (0, _cepPromise.cep) is not a function. (In '(0, _cepPromise.cep)(86055645, {
                providers: ['correios']
              })', '(0, _cepPromise.cep)' is undefined)
```

Criando um async, funciona. Portanto, acredito que o cep-promise deveria ser rearranjado para funcionar com async/await diretamente.

```ts
    async function CEP(CEP: string) {
      return new Promise(resolve => {
        cep(CEP, {timeout: 5000, providers: ['brasilapi']}).then(response => resolve(response));
      });
    }

    const cepResponse = await CEP(userInfo.postal_code);
    console.log('cepResponse ', cepResponse);
```

**Comment by bearkfear** [comment]
Poderia dar mais contexto, a função superior que você está usando tem o prefixo async?

**Comment by hadnet** [comment]
Sim.

**Comment by isaiasvallejos** [comment]
Acredito que seja algum problema de `import` ou de compilação do React, porque acho que em teoria o `cep-promise` aceita async/await pois retorna uma `Promise`.

Consegue mostrar como você está importando?

**Comment by felinto-dev** [comment]
Acho que deveria ser

`return cep(CEP, {timeout: 5000, providers: ['brasilapi']})...
`
em vez

`cep(CEP, {timeout: 5000, providers: ['brasilapi']})...
`
não?

**Comment by felinto-dev** [comment]
Estou com o mesmo problema:
https://github.com/BrasilAPI/cep-promise/issues/212

Na resposta da issue coloquei que achei uma solução, no entanto, estou usando um framework chamado Nest e lá não funciona import de jeito nenhum, mesmo usando o mesmo tsconfig.json

Lá tenho que fazer
`const cepPromise = require("cep-promise")`

Infelizmente não tenho conhecimento sobre isso para fazer uma PR, mas o suporte ao TypeScript, bem como error handling do módulo está muito fraco, porque:
1. Devido a esses problemas de compatibilidade com o TypeScript de usar import diretamente.
2. Quando você digita um cep inválido, mesmo com 8 caracteres, ele retorna um erro dizendo que nenhum provedor encontrou aquele cep. Acho interessante continuar tendo essa opção de retornar um erro, mas nas opções do cep promise, poderia ter a opção de especificar de somente retornar NULL. Acho muito mais legível fazer isso:

```ts
const cepData = await cep('58308000', { onError: 'RETURN_NULL' })
if (!cepData) {
    ...envia uma mensagem para o usuário
}
```

do que isso

```ts
const cepData = cep('58308000')
    .then((res) => console.log(res))
    .catch(err) => console.log('VOCÊ TRATA O ERRO NO BLOCO CATCH'))
```

**Comment by hadnet** [comment]
> Acho que deveria ser
> 
> `return cep(CEP, {timeout: 5000, providers: ['brasilapi']})... `
> em vez
> 
> `cep(CEP, {timeout: 5000, providers: ['brasilapi']})... `
> não?

Sim, mas desse jeito TS dispara erro pra tudo quanto é lado. Por isso digo que tem que ser rearranjado.

**Comment by bearkfear** [comment]
Acredito que a conversão para typescript será muito benéfica, se uma boa galerinha quiser eu posso fazer o PR. #169
